plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -107,7 +109,10 @@ applyAndBetaReduce rhs args0 = do
         -- the body `a` will be beta reduced in
         Term tyname name uni fun ann ->
         InlineM tyname name uni fun ann Bool
-      safeToBetaReduce = shouldUnconditionallyInline Strict
+      safeToBetaReduce arg term =
+        -- Does not inline constants because term size might increase
+        -- and other inlining optimisations will not get applied.
+        local (iiInlineConstants .~ False) . shouldUnconditionallyInline Strict arg term
```

Comment:
Turning it back to `True` will actually _decrease_ the size of the term in all the tests which fail.

Reply:
Actually, this doesn't work, and I can't think of a nice and quick way to implement this. The idea was that since callsite inlining _requires_ that the resulting term is smaller in size, having constant inlining turned on by default might not allow the callsite inlining to take place (if my reading of the code is correct).

Now, it seems like our test suites don't contain any examples where constant inlining increases program size. Can we be sure that the test suite reflects actual real world behavior? Should we try to add some more test cases?

Reply:
> having constant inlining turned on by default might not allow the callsite inlining to take place

I don't think it can. The size is measured by `termSize`, which is the number of AST nodes, and is only an approximation of the actual script size. In particular, a constant and a variable are both one node, even if it's a big constant. While it is not precise, it means inlining a constant cannot increase `termSize`.

Reply:
Right, then the results make sense. Thank you.